rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/370_profile.yml

```diff
@@ -220,8 +220,6 @@ dfs knn vector profiling:
   - gt: { profile.shards.0.dfs.knn.0.query.0.breakdown.score: 0 }
   - match: { profile.shards.0.dfs.knn.0.query.0.breakdown.compute_max_score_count: 0 }
   - match: { profile.shards.0.dfs.knn.0.query.0.breakdown.compute_max_score: 0 }
-  - gt: { profile.shards.0.dfs.knn.0.query.0.breakdown.advance_count: 0 }
-  - gt: { profile.shards.0.dfs.knn.0.query.0.breakdown.advance: 0 }
```

Comment:
This: https://github.com/apache/lucene/pull/12481

Seems to have broken profiling. Digging into all the profiling code, it seems like `advance` is no longer being called on our scorer iterators after this change.

Looks like it calls `nextDoc`, and doesn't forward the call to `advance(docId + 1)` any longer all the time. Seems ok to me.